src/tunnel.cc

```diff
@@ -441,6 +478,18 @@ TunnelStateData::~TunnelStateData()
     delete savedError;
 }
 
+TunnelStateData::Connection::Connection(const char * const aSide):
+    len(0),
```

Comment:
Most of these data members should be initialized in their class declarations instead, but that requires more out-of-scope changes, especially since changing initialization of some of those data members should be combined with fixing their _type_. I believe those improvements deserve a dedicated PR (or two). This PR preserves existing initialization approach (and types) for existing data members. I de-inlined this constructor definition (because it had to be changed) to reduce class declaration noise and to simplify its source code formatting.

This comment does not request any PR changes.